integration-tests/singleton_flush_test.go

```diff
@@ -0,0 +1,237 @@
+package main
+
+import (
+	"encoding/xml"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"gopkg.in/yaml.v2"
+)
+
+type SingletonFlushTestSuite struct {
+	suite.Suite
+	ServiceID    string
+	ServiceToken string
+	AppIDs       []string
+	PlanID       string
+	Metrics      []Metric
+}
+
+func (suite *SingletonFlushTestSuite) SetupSuite() {
+	suite.ServiceID = "test-service-id"
+	suite.AppIDs = []string{"test-app-id-1", "test-app-id-2"}
+	suite.ServiceToken = "test-service-token"
+	suite.PlanID = "test-plan-id"
+	suite.Metrics = []Metric{
+		{"hits", "1", []UsageLimit{
+			{Day, 10000},
+		}},
+		{"rq", "2", []UsageLimit{
+			{Month, 10000},
+		}},
+	}
+
+}
+
+func (suite *SingletonFlushTestSuite) BeforeTest(suiteName, testName string) {
+	if testName == "TestSingletonContainerFlush" {
+		configErr := configureSingletonFlush("ContainerLimit")
+		require.Nilf(suite.T(), configErr, "Error configuring envoy.yaml for container flush : %v", configErr)
+
+		require.Eventually(suite.T(), func() bool {
+			fmt.Printf("Creating service with service_id: %s, service_token:%s", suite.ServiceID, suite.ServiceToken)
+			serviceErr := CreateService(suite.ServiceID, suite.ServiceToken)
+			if serviceErr != nil {
+				return false
+			}
+			return true
```

Comment:
How about `Push()`?

Reply:
Yeah will change to those in the next PR with the other test cases for the singleton.